js/playlist.js

```diff
@@ -1,18 +1,20 @@
 define(['ytDataAPI', 'ytIframeAPI', 'template','jquery'],
 function(ytData,ytIframe,template,$) {
     var playlist=[];
+    var durations=[];
     var total_duration = 0;
     var playlistChangeListeners = []
-
+    var curr_id;
     var setup = function() {
+        console.log($.query);
         // Load playlist from cache/url
         if($.query.playlist!=undefined){
             ytData.api.videos.list({
-                id:$.query.playlist,
+                id:$.query.playlist[curr_id],
```

Comment:
Firstly, i don't see where you have changed the share link generation code to reflect as list of playlists. So I don't think this will work. I know my project does not have tests, but you should ensure as a rule of thumb that you once manually test every line of change that you make. Remember that this is a URL param, so this will be a list only if it is set in a specific way.

Secondly, the user generally wouldn't want to share each one of his playlists, only the current one. So the change here should rather be that `$.query.playlist` is added to the user's list of playlists as a new temp playlist.